core/src/com/unciv/logic/automation/Automation.kt

```diff
@@ -64,33 +65,46 @@ object Automation {
         }
 
         val surplusFood = city.cityStats.currentCityStats[Stat.Food]
+        val starving = surplusFood < 0
         // If current Production converts Food into Production, then calculate increased Production Yield
         if (cityStatsObj.canConvertFoodToProduction(surplusFood, city.cityConstructions.getCurrentConstruction())) {
             // calculate delta increase of food->prod. This isn't linear
             yieldStats.production += cityStatsObj.getProductionFromExcessiveFood(surplusFood+yieldStats.food) - cityStatsObj.getProductionFromExcessiveFood(surplusFood)
             yieldStats.food = 0f  // all food goes to 0
         }
+
+        var growthFood =
+            yieldStats.food  // amount of food yield beyond needed to avoid starving, default all
+        if (starving)
+            growthFood = max(yieldStats.food + surplusFood, 0f)
```

Comment:
Please use .coerceAtLeast(0f) because in my opinion (and that of others I've talked to about this) it makes it much clearer what we're aiming for